src/Raven.Server/Utils/ThreadsUsage.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using Raven.Server.Dashboard;
+using Sparrow.Logging;
+using Sparrow.Utils;
+using ThreadState = System.Diagnostics.ThreadState;
+
+namespace Raven.Server.Utils
+{
+    public class ThreadsUsage
+    {
+        private static readonly Logger Logger = LoggingSource.Instance.GetLogger<MachineResources>("ThreadsUsage");
+        private (long TotalProcessorTimeTicks, long TimeTicks) _processTimes;
+        private Dictionary<int, long> _threadTimesInfo = new Dictionary<int, long>();
+
+        public ThreadsUsage()
+        {
+            using (var process = Process.GetCurrentProcess())
+            {
+                _processTimes = CpuUsage.GetProcessTimes(process);
+            }
+
+            // calculate the initial threads usage
+            Thread.Sleep(10);
```

Comment:
why?

Reply:
Isn't there a better way to do that rather than force a sleep?

Reply:
done